### PR TITLE
Refactor assessment archiving to support updating to AssessmentModel version 0.10.1

### PR DESCRIPTION
**Pull request by nategbrown9**

### CODE REVIEW

File: assessmentmodel-sdk/src/main/java/org/sagebionetworks/bridge/assessmentmodel/upload/AssessmentArchiver.kt [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+            .withPhoneInfo(bridgeConfig.deviceName)
+    }
+
+    fun buildArchive() : Archive {
+        // Iterate through all the results within this collection and add if they are `JsonFileArchivableResult`.
+        addBranchResults(assessmentResult)
+
+        //Add assessment result file
+        Logger.d("Writing result for assessment ${assessmentResult.identifier}")
+        val resultString =
```
Comment (syoung-smallwisdom) [inline assessmentmodel-sdk/src/main/java/org/sagebionetworks/bridge/assessmentmodel/upload/AssessmentArchiver.kt]: ?

File: assessmentmodel-sdk/src/main/java/org/sagebionetworks/bridge/assessmentmodel/upload/AssessmentResultArchiveUploader.kt [hunk 2]
```diff
@@ -94,7 +67,7 @@ abstract class AssessmentResultArchiveUploader(
         )
 
 
-        val uploadFile = persist(assessmentRunUUID, builder.build(), uploadMetadata, sessionWindowExpiration)
+        val uploadFile = persist(assessmentRunUUID, archiver.buildArchive(), uploadMetadata, sessionWindowExpiration)
```
Comment (syoung-smallwisdom) [inline assessmentmodel-sdk/src/main/java/org/sagebionetworks/bridge/assessmentmodel/upload/AssessmentResultArchiveUploader.kt:70]: is this all on a background thread already?

**Review by syoung-smallwisdom — APPROVED** [review]
Looks good

One ? about what looks like dangling code

File: assessmentmodel-sdk/src/main/java/org/sagebionetworks/bridge/assessmentmodel/upload/AssessmentResultArchiveUploader.kt (hunk 2 above)
Comment (nategbrown9) [reply assessmentmodel-sdk/src/main/java/org/sagebionetworks/bridge/assessmentmodel/upload/AssessmentResultArchiveUploader.kt:70]: The archiving code is all run on a background thread by the app's view model.

**Comment by nategbrown9** [comment]
Created a develop branch to merge this PR into.

**Pull request merged**